subprojects/test-kit/src/integTest/groovy/org/gradle/testkit/runner/GradleRunnerGradleVersionIntegrationTest.groovy

```diff
@@ -30,20 +30,18 @@ import spock.lang.Shared
 import static org.gradle.integtests.fixtures.RetryConditions.onIssueWithReleasedGradleVersion
 
 @NonCrossVersion
-@Requires(TestPrecondition.ONLINE)
+@Requires(value = TestPrecondition.ONLINE, adhoc = { BaseGradleRunnerIntegrationTest.findMinimumSupportedGradleVersion() != null })
```

Comment:
IIUC, we don't need this `adhoc` because we already throw `AssumptionViolationException` in `getMinimumSupportedGradleVersion()`, right?

Reply:
No, because `@Retry` causes problems:
<details>
<summary>Stacktrace</summary>

```
org.spockframework.runtime.extension.ExtensionException: Failed to evaluate @Retry condition
	at app//org.spockframework.runtime.extension.builtin.RetryBaseInterceptor.satisfiesCondition(RetryBaseInterceptor.java:82)
	at app//org.spockframework.runtime.extension.builtin.RetryBaseInterceptor.isExpected(RetryBaseInterceptor.java:57)
	at app//org.spockframework.runtime.extension.builtin.RetryBaseInterceptor.handleInvocation(RetryBaseInterceptor.java:93)
	at app//org.spockframework.runtime.extension.builtin.RetryFeatureInterceptor.intercept(RetryFeatureInterceptor.java:35)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at app//org.spockframework.junit4.ExceptionAdapterInterceptor.intercept(ExceptionAdapterInterceptor.java:13)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:102)
	at app//org.gradle.test.fixtures.file.CleanupTestDirectoryExtension$FailureCleanupInterceptor.intercept(CleanupTestDirectoryExtension.groovy:57)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at app//org.spockframework.junit4.ExceptionAdapterInterceptor.intercept(ExceptionAdapterInterceptor.java:13)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at app//org.spockframework.junit4.AbstractRuleInterceptor$1.evaluate(AbstractRuleInterceptor.java:46)
	at app//org.gradle.util.SetSystemProperties$1.evaluate(SetSystemProperties.java:53)
	at app//org.gradle.test.fixtures.file.AbstractTestDirectoryProvider$TestDirectoryCleaningStatement.evaluate(AbstractTestDirectoryProvider.java:101)
	at app//org.spockframework.junit4.AbstractRuleInterceptor.evaluateStatement(AbstractRuleInterceptor.java:66)
	at app//org.spockframework.junit4.TestRuleInterceptor.intercept(TestRuleInterceptor.java:43)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at app//org.gradle.integtests.fixtures.extensions.AbstractMultiTestInterceptor$ParameterizedFeatureMultiVersionInterceptor.interceptIterationExecution(AbstractMultiTestInterceptor.java:231)
	at app//org.spockframework.runtime.extension.AbstractMethodInterceptor.intercept(AbstractMethodInterceptor.java:51)
	at app//org.spockframework.runtime.extension.MethodInvocation.proceed(MethodInvocation.java:101)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask$DefaultDynamicTestExecutor.execute(NodeTestTask.java:226)
	at app//org.gradle.integtests.fixtures.extensions.AbstractMultiTestInterceptor.lambda$interceptFeature$2(AbstractMultiTestInterceptor.java:84)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base@19/java.util.ArrayList.forEach(ArrayList.java:1511)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:156)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base@19/java.util.ArrayList.forEach(ArrayList.java:1511)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at app//org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at app//org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at app//org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at app//org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.LauncherMain$c.a(SourceFile:192)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.LauncherMain$c.b(SourceFile:183)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.LauncherMain$c.a(SourceFile:175)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.LauncherMain.a(SourceFile:79)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.LauncherMain.a(SourceFile:70)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.k.a(SourceFile:20)
	at app//com.gradle.enterprise.testdistribution.launcher.forked.LauncherMain.main(SourceFile:55)
Caused by: org.junit.AssumptionViolatedException: No version of Gradle supports Java 19
	at app//org.gradle.testkit.runner.BaseGradleRunnerIntegrationTest.getMinimumSupportedGradleVersion(BaseGradleRunnerIntegrationTest.groovy:210)
	at app//org.gradle.testkit.runner.GradleRunnerGradleVersionIntegrationTest.getDaemonsFixture(GradleRunnerGradleVersionIntegrationTest.groovy:44)
	at app//org.gradle.integtests.fixtures.RetryConditions.onIssueWithReleasedGradleVersion(RetryConditions.groovy:74)
	at app//org.gradle.testkit.runner.GradleRunnerGradleVersionIntegrationTest$_closure1.doCall(GradleRunnerGradleVersionIntegrationTest.groovy:34)
	at app//groovy.lang.Closure.call(Closure.java:412)
	at app//groovy.lang.Closure.call(Closure.java:406)
	at app//org.spockframework.runtime.extension.builtin.RetryBaseInterceptor.satisfiesCondition(RetryBaseInterceptor.java:80)
	... 78 more
```
</details>